### ISSUE TITLE
save_pod_item() don't stop validation after the first "error" lets collect em

### ISSUE THREAD

**Issue by quasel** [pods-framework/pods#2670]
If i am not getting something wrong save_pod_item() currently returns an pods_error()  in the foreach ( $fields_active as $field ) loop on the first validation error it hits.

suggestion: Validate all fields and return all errors together

could improve error handling for the JSON API and any other Client - for errors wich are not caught / handled by client side validation...

**Comment by jamesgol** [comment]
Agreed.  For pods forms the javascript is supposed to catch some of the validation (required fields), but that won't happen on JSON API or other stuff.  Not sure how other integrations like Caldera Forms and Gravity forms work.

**Comment by Shelob9** [comment]
I'm putting this on 3.0 milestone as it may be handled by moving to CMB2 for validation, though not sure how that deals with saving via internal API or JSON API--which is the real issue here--that our validation done via JavaScript needs to be done when saved via an API.

**Comment by jamesgol** [comment]
Was thinking about this a bit this morning.  If we don't want to completely overhaul pods_error then the best option might be to let pods_error accept a WP_Error as $error.  Then in the loop in save_pod_item have it create a new WP_Error and call ->add for each one.  After the loop check that var and call pods_error on it.

Just some random rambling, when I have some free time I'll play around with it.

**Comment by jimtrue** [comment]
Also related to #4231 and #1106
